src/fmpi2c.rs

```diff
@@ -64,25 +64,33 @@ where
     }
 }
 
-impl PinScl<FMPI2C1> for gpioc::PC6<AlternateOD<4>> {}
-impl PinSda<FMPI2C1> for gpioc::PC7<AlternateOD<4>> {}
-impl PinSda<FMPI2C1> for gpiob::PB3<AlternateOD<4>> {}
-impl PinScl<FMPI2C1> for gpiob::PB10<AlternateOD<9>> {}
-impl PinSda<FMPI2C1> for gpiob::PB14<AlternateOD<4>> {}
-impl PinScl<FMPI2C1> for gpiob::PB15<AlternateOD<4>> {}
-impl PinScl<FMPI2C1> for gpiod::PD12<AlternateOD<4>> {}
-impl PinScl<FMPI2C1> for gpiob::PB13<AlternateOD<4>> {}
-impl PinScl<FMPI2C1> for gpiod::PD14<AlternateOD<4>> {}
-impl PinScl<FMPI2C1> for gpiod::PD15<AlternateOD<4>> {}
-impl PinScl<FMPI2C1> for gpiof::PF14<AlternateOD<4>> {}
-impl PinScl<FMPI2C1> for gpiof::PF15<AlternateOD<4>> {}
-
-impl<SCL, SDA> FMPI2c<FMPI2C1, (SCL, SDA)>
+macro_rules! pin {
+    ($trait:ident<$I2C:ident> for $gpio:ident::$PX:ident<$A:literal>) => {
+        impl<MODE> $trait<$I2C> for $gpio::$PX<MODE> {
+            type A = Const<$A>;
+        }
+    };
+}
+
+pin!(PinScl<FMPI2C1> for gpioc::PC6<4>);
+pin!(PinSda<FMPI2C1> for gpioc::PC7<4>);
+pin!(PinSda<FMPI2C1> for gpiob::PB3<4>);
+pin!(PinScl<FMPI2C1> for gpiob::PB10<9>);
+pin!(PinSda<FMPI2C1> for gpiob::PB14<4>);
+pin!(PinScl<FMPI2C1> for gpiob::PB15<4>);
+pin!(PinScl<FMPI2C1> for gpiod::PD12<4>);
+pin!(PinScl<FMPI2C1> for gpiob::PB13<4>);
+pin!(PinScl<FMPI2C1> for gpiod::PD14<4>);
+pin!(PinScl<FMPI2C1> for gpiod::PD15<4>);
+pin!(PinScl<FMPI2C1> for gpiof::PF14<4>);
+pin!(PinScl<FMPI2C1> for gpiof::PF15<4>);
+
+impl<SCL, SDA, const SCLA: u8, const SDAA: u8> FMPI2c<FMPI2C1, (SCL, SDA)>
 where
-    SCL: PinScl<FMPI2C1>,
-    SDA: PinSda<FMPI2C1>,
+    SCL: PinScl<FMPI2C1, A = Const<SCLA>> + SetAlternateOD<SCLA>,
+    SDA: PinSda<FMPI2C1, A = Const<SDAA>> + SetAlternateOD<SDAA>,
 {
-    pub fn new<M: Into<FmpMode>>(i2c: FMPI2C1, pins: (SCL, SDA), mode: M) -> Self {
+    pub fn new<M: Into<FmpMode>>(i2c: FMPI2C1, mut pins: (SCL, SDA), mode: M) -> Self {
```

Comment:
Since you've been mentioning breaking changes in the CHANGELOG, why is this not considered a breaking change?

Reply:
Are you asking about all PR? If so, I've not found it breaks something that had worked earlier.

If you are about  this `mut`. It is internal `mut`. It does not affect on API.

Reply:
Hm, it is a public method so someone could use it, no?

Reply:
This is not type `mut`. This is value `mut`. Inside function.

Reply:
```rust
fn new(mut pins: (SCL, SDA)) {
    // do something with `pins`
}
```
is just shorthand for:
```rust
fn new(pins: (SCL, SDA)) {
    let mut pins = pins;
    // do something with `pins`
}
```

Reply:
🤦‍♂️ Somehow I was seeing a reference there. Never mind.